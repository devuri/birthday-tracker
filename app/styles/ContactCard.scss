contactCard {
  background-color: #09C6AB;
  border-bottom-width: 0.5;
  border-color: lighten(#013240, 15%);
  flex: 1;
  padding: 10px 0 10px 10px;
  width: 320px;
}

nameText {
  color: #013240;
  font-size: 16px;
  font-weight: bold;
  padding:0 0 5px 20px;
}

birthdayText {
  color: #013240;
}

row {
  flex-direction: row;
}

card {
  flex: 1;
  justify-content: center;
  margin-left: 20px;
}

listAvatar {
  border-radius: 37.5px;
  height: 75px;
  width: 75px;
}
