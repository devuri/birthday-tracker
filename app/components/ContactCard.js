'use strict';
import React, { Component } from 'react';
import styles from '../styles/main';
import { View, Text, TouchableHighlight, Image } from 'react-native';
import { Actions } from 'react-native-router-flux';

const ContactCard = ({ firstName, lastName, birthdayDate, id, onPress, avatar }) => {
  console.log(avatar);
  return (
    <TouchableHighlight
      style={styles.contactCard}
      onPress={() => Actions.contactInfo({firstName, lastName, birthdayDate, id, onPress, avatar })}>
      <View style={styles.row}>
        <View>
          { !avatar ? <Image style={styles.listAvatar} source={require('../img/list-avatar.png')} /> :
            <Image style={styles.listAvatar} source={{uri: avatar.uri}} />
          }
        </View>
        <View style={styles.card}>
          <Text style={styles.nameText}>{`${firstName} ${lastName}`}</Text>
          <Text style={styles.birthdayText}>ðŸŽˆ{birthdayDate}</Text>
        </View>
      </View>
    </TouchableHighlight>
  );
}

export default ContactCard;
