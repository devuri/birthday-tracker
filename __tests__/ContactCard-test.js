import 'react-native';
import React from 'react';
import ContactCard from '../app/components/ContactCard.js';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <ContactCard  />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
