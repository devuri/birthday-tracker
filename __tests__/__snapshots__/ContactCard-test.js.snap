exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "backgroundColor": "#09C6AB",
        "borderBottomWidth": 0.5,
        "borderColor": "#026d8b",
        "flex": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 0,
        "paddingTop": 10,
        "width": 320,
      },
    ]
  }
  testID={undefined}>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }>
    <View>
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 37,
            "height": 75,
            "width": 75,
          }
        } />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "marginLeft": 20,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#013240",
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingBottom": 5,
            "paddingLeft": 20,
            "paddingRight": 0,
            "paddingTop": 0,
          }
        }>
        undefined undefined
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#013240",
          }
        }>
        🎈
      </Text>
    </View>
  </View>
</View>
`;
